### ISSUE TITLE
The task committed to yarn is not monitored for short-circuit Read

### ISSUE THREAD

**Issue by ljk-1993** [Alluxio/alluxio#12617]
**Alluxio Version:**
v2.3.0
**Describe the bug**
All tasks committed to YARN are not monitored for short-circuit Read.

**To Reproduce**

First,after we submitted a Spark SQL task for many times, alluxio's monitoring information always showed that short-circuit Read was 0B. After we performed a query task in the Hive client, short-circuit Read had monitoring data. However, after we submitted the same Spark SQL task again, the value of short-circuit Read did not change.

Therefore, we add wo debug logs in alluxio.client. Block. Stream. BlockInStream.create:

The first is to print log.warn ("-- ljk-debug---- create stream for short-circuit read!") before the return createLocalBlockInStream(context, dataSource, blockId, blockSize, options). );

The second is to output log.warn ("-- lJk-debug ----create stream for Alluxio Remote Read!") before return createGrpcBlockInStream(Context, dataSource, dataSourceType, Build.buildPartial (),blockSize, options). );

Then we examine the applicationlog.The application log in YARN shows' WARN BlockInStream: -- LJk-debug --create stream for short-circuit read! '

**Expected behavior**

Alluxio's monitoring information should show how much data is loaded into the application via short-circuit Read, and the performance of the application should be improved.

**Urgency**

emergency

**Comment by gpang** [comment]
Hi @ljk-1993 
Can you set this client property? `alluxio.user.metrics.collection.enabled=true`

**Comment by ljk-1993** [comment]
@gpang 
We have set this parameter on all alluxio nodes. Need to set this parameter in the program?

**Comment by ljk-1993** [comment]
We specified parameters when we submitted the Spark task:
--conf 'spark.driver.extraJavaOptions=-Dalluxio.user.metrics.collection.enabled=true' 
--conf 'spark.executor.extraJavaOptions=-Dalluxio.user.metrics.collection.enabled=true' 

Alluxio can monitor that the data is loaded by short-circuit Read.

I found in the code alluxio.user.metrics.collection.enabled the value is false, if the default value is set to true will affect performance?

```
  public static final PropertyKey USER_METRICS_COLLECTION_ENABLED =
      new Builder(Name.USER_METRICS_COLLECTION_ENABLED)
          .setDefaultValue(false)
          .setDescription("Enable collecting the client-side metrics and heartbeat them to master")
          .setConsistencyCheckLevel(ConsistencyCheckLevel.WARN)
          .setScope(Scope.CLIENT)
          .build();
```

**Comment by gpang** [comment]
@ljk-1993 Do you know if this happens even if you do not use spark? For example, will this happen even if you use something like the Alluxio shell and copyFromLocal?

If so, can you show a simple set of steps for others to reproduce?

**Comment by ljk-1993** [comment]
@gpang 
the default value of  alluxio.user.metrics.collection.enabled is set to true.
When I submitted Spark task, I did not specify the parameter mentioned above. Alluxio can monitor that the data is loaded by short-circuit Read.

I have uploaded a local file to the worker1.bigdata node via copyFromLocal. Alluxio monitoring shows that data is written to the Alluxio cluster via short-circuit Write.

After I executed the 'bin/alluxio fs cat' command on the worker1.bigdata, Alluxio was able to monitor that the data was loaded by short-circuit Read
